src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -317,6 +322,64 @@ public Response replaceFileInDataset(
         }
             
     } // end: replaceFileInDataset
+
+    /**
+     * Delete an Existing File 
+     * 
+     * @param id file ID or peristent ID
+     */
+    @DELETE
+    @Path("{id}")
+    public Response deleteFileInDataset(@PathParam("id") String fileIdOrPersistentId){
+
+        if (!systemConfig.isHTTPUpload()) {
+            return error(Response.Status.SERVICE_UNAVAILABLE, BundleUtil.getStringFromBundle("file.api.httpDisabled"));
+        }
+        // (1) Get the user from the API key
+        User authUser;
+        try {
+            authUser = findUserOrDie();
+        } catch (AbstractApiBean.WrappedResponse ex) {
+            return error(Response.Status.FORBIDDEN, BundleUtil.getStringFromBundle("file.addreplace.error.auth"));
+        }
+
+        msg("DELETE!");
+
+        // (2) Delete
+        boolean deletePhysicalFile = false;
+        try {
+            DataFile dataFile = findDataFileOrDie(fileIdOrPersistentId);
+            List<FileMetadata> fileToDelete = dataFile.getFileMetadatas();
+            Dataset dataset = dataFile.getOwner();
+            DatasetVersion v = dataset.getOrCreateEditVersion();
+            DataverseRequest dvRequest2 = createDataverseRequest(authUser);
```

Comment:
If we do use createDataverseRequest above we shouldn't need it again here.

Reply:
I have moved the DataverseRequest creation to just after the user retrieval.